ceno_zkvm/src/instructions/riscv/ecall/keccak.rs

```diff
@@ -98,11 +101,15 @@ impl<E: ExtensionField> Instruction<E> for KeccakInstruction<E> {
             E::BaseField::ZERO.expr(),
         ))?;
 
+        let mut layout = <KeccakLayout<E> as gkr_iop::ProtocolBuilder<E>>::build_layer_logic(
```

Comment:
Flow of keccak opcode 
1. build circuit: register, state constrain
2. call `gkr_iop::build_layer_logic` to build gkr iop related logic
3. take input32/output32 expression from gkr-iop layout and build memory constrain
4. get chip, out_evals from layout.finalized
5. get layer and add to chip